### DISCUSSION TITLE
Remote code editing (solo)

### DISCUSSION THREAD

**Discussion by iamvoidcoder** [zed-industries/zed#6579] [Feature Requests]
Hi

Not totally sure I'm placing this in the right discussions category.

I landed on zed.dev page by accident. Sort of built a habit for periodically googling around to see if someone finally came up with a decent editor for remote code editing on the server. There are plenty of cases where you have to deal with a large code base in-place on the server, as opposed to working on a local copy then syncing changes to the server. 

The traditional approach has always been vim/emacs/whatever text mode editor in SSH session. Something I always hated but had to live with because of no alternatives. Always dreamed of a powerful GUI editor with seamless remote project access.

I'm not counting here solutions such as locally mounting remote file system via sshfs/nfs/whatever. Nor real-time syncing of local and remote file system solutions. Nor various half-backed file access plugins for some established GUI editors working over ssh, scp, sftp and what not. While this works for small scale projects or granular small edits, none of that works for full time code editing on large scale remote code base with GBs of code. Not useable for full remote project wide indexing, IntelliSense / LSP, full project wide searches etc. This clearly needs a server side backend of sorts, not doable from a local client by remotely accessing individual files as it is prohibitively slow on large code bases.

So this morning I was on my usual routine google hunt for an editor with good remote editing caps and accidentally discovered Zed. Started reading and at first it sounded like I finally hit the jackpot! But then I realised, while it seems to have the remote project code editing features I want, this is primarily for team colab rather than remote project code editing. Which is a bit strange as I think the two task are closely related in a way. At least in my environment there is more hunger primarily for a good remote editor than team colab, though ideally these are a single integrated solution.

I suppose one can run an instance of Zed on the server under some non-existent virtual team member and join it remotely as a teammate from your local Zed instance. Though I suppose this will need a window manger on the server & remote desktop and generally sounds like an overkill. 

I wonder whether there is a way to just run some sort of headless Zed backend on the code server that yourself and teammates can connect to for remote code editing and the rest of Zed goodies? Or would you possibly consider adding such a feature in?

Thanks

**Comment by jansol** [comment]
This feature has been discussed at length in #130

**Reply by iamvoidcoder** [reply]
Thanks. I run a search through discussions before posting but somehow managed to overlook #130. Though I can't see any input from Zed Industries on the topic. 

Still useful discussion. I tried JetBrains Gateway SSH solution at some point and it was super slow, practically unusable for a large code base project. 

Didn't know VS Code had that too. Don't like it is as IDE but definitely will give the remote feature a try.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
